presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java

```diff
@@ -102,7 +102,7 @@ public PrestoCliTests()
     }
 
     @AfterTestWithContext
-    public void stopPresto()
+    public synchronized void stopPresto()
```

Comment:
We should have a better way to do this. The intent is not immediately clear when reading the code and it's easy to miss when adding new test cases. Additionally, it ties up all the test threads (maybe not a problem if we only run one test class at once).

It would be better if product tests supported `@Test(singleThreaded = true)` or similar at the class level.